plugins/i3s/io/EsriReader.cpp

```diff
@@ -0,0 +1,435 @@
+        -shared resources: <node-url>/shared/
+        -feature data: <node-url>/features/<feature-data-bundle-id>
+        -geometry data: <node-url>/geometries/<geometry-data-bundle-id>
+        -texture data: <node-url>/textures/<texture-data-bundle-id>
+        */
+
+        //Build the node list that will tell us which nodes overlap with bounds
+        std::vector<int> nodes;
+        buildNodeList(nodes, 0);
+
+        //Create view with overlapping leaf nodes
+        //Will create a thread pool on the createview class and iterate
+        //through the node list for the nodes to be pulled.
+        std::cout << "Fetching binaries" << std::endl;
+        Pool p(m_args.threads);
+        for (std::size_t i = 0; i < nodes.size(); i++)
+        {
+            std::cout << "\r" << i << "/" << nodes.size();
+            std::cout.flush();
+            std::string localUrl;
+
+            localUrl = m_filename + "/nodes/" + std::to_string(nodes[i]);
+
+            p.add([localUrl, this, &view]()
+            {
+                createView(localUrl, view);
+            });
+        }
+        const std::size_t pointSize(view->layout()->pointSize());
+        return view->size();
+    }
+
+    // Create the BOX3D for the node. This will be used for testing overlap.
+    BOX3D EsriReader::parseBox(Json::Value base)
+    {
+        // Pull XYZ in lat/lon.
+        Json::Value center = base["obb"]["center"];
+        double x = center[0].asDouble();
+        double y = center[1].asDouble();
+        double z = center[2].asDouble();
+
+        // Convert to ECEF so the OBB offsets in meters will match up.
+        OCTTransform(m_toEcefTransform, 1, &x, &y, &z);
+
+        Json::Value hsize = base["obb"]["halfSize"];
+        const double hx = hsize[0].asDouble();
+        const double hy = hsize[1].asDouble();
+        const double hz = hsize[2].asDouble();
+
+        Json::Value quat = base["obb"]["quaternion"];
+        const double w = quat[0].asDouble();
+        const double i = quat[1].asDouble();
+        const double j = quat[2].asDouble();
+        const double k = quat[3].asDouble();
+
+        // Create quat object and normalize it for use
+        Eigen::Quaterniond q(w, i, j, k);
+        q.normalize();
+
+        // Here we'll convert our halfsizes to x, y, and z vectors in respective
+        // directions, which will give us new bounding planes
+        Eigen::Vector3d vxmax(hx,   0,   0);
+        Eigen::Vector3d vymax(0,   hy,   0);
+        Eigen::Vector3d vzmax(0,    0,  hz);
+
+        // Create quaternion-like vectors
+        Eigen::Quaterniond pxmax, pxmin, pymax, pymin, pzmax, pzmin;
+        pxmax.w() = 0;
+        pymax.w() = 0;
+        pzmax.w() = 0;
+        pxmax.vec() = vxmax;
+        pymax.vec() = vymax;
+        pzmax.vec() = vzmax;
+
+        // Rotate all the individual vectors
+        // gives us offset for the of the new x/y/zmax/min planes
+        Eigen::Quaterniond rxmax = q * pxmax * q.inverse();
```

Comment:
These should be named better.  They aren't "max" values, they are XYZ vector of OBB-half-size in a transformed coordinate space.  Maybe `xPrime` or `xTransformed`, for example?